overcooked_ai_py/agents/agent.py

```diff
@@ -515,3 +532,2155 @@ def ml_action(self, state):
+                                                                      current_goal='use_dish')
+            elif greedy_goal == None:
+                raise ValueError('greedy_goal not specified')
+
+        return motion_goals
+
+    def find_min_cost_of_achieving_goal(self, player, default_motion_goals, other_player, state):
+        # Remove invalid goals:
+        own_motion_goals = list(
+            filter(lambda goal: self.mlp.mp.is_valid_motion_start_goal_pair(player.pos_and_or, goal),
+                   default_motion_goals))
+        others_motion_goals = list(
+            filter(lambda goal: self.mlp.mp.is_valid_motion_start_goal_pair(other_player.pos_and_or, goal),
+                   default_motion_goals))
+        # Find costs:
+        own_min_cost = self.find_min_plan_cost(own_motion_goals, state, self.agent_index)
+        others_min_cost = self.find_min_plan_cost(others_motion_goals, state, 1 - self.agent_index)
+
+        return own_min_cost, others_min_cost
+
+    # Functions specific to each part of the decision tree:
+
+    def revise_pickup_onion_goal_considering_other_player_and_noise(self, number_of_pots, player,
+                    default_motion_goals, other_player, state, other_has_onion, counter_objects, am, temp_dont_drop):
+        """Revise what the best goal is, depending on whether the other player is in a better/worse position to
+        achieve the default goal. Noise refers to making wrong decisions with a given prob.
+        Note: If there are two pots, then we can only get inside this 'if' when both pots have onion first on the
+        list of tasks to do, so we can always get an onion. For 1 pot then just look what's next on the list"""
+
+        # ASSUMING TWO POTS MAX THROUGHOUT
+        if number_of_pots == 1:
+
+            own_min_cost, others_min_cost = self.find_min_cost_of_achieving_goal(player,
+                                                                                 default_motion_goals, other_player,
+                                                                                 state)
+
+            # Get the onion if the other player doesn't have an onion or they do have an onion but are further away
+            if (not other_has_onion) and (
+                    others_min_cost >= own_min_cost):  # TODO: make it random if others_min_cost = own_min_cost ?
+                logging.info('Getting an onion: assume in better position that other player')
+                motion_goals = am.pickup_onion_actions(state, counter_objects, self.only_take_dispenser_onions)
+                motion_goals = self.sometimes_make_wrong_decision(motion_goals, state, counter_objects, am,
+                                                                  current_goal='onion')
+
+            elif other_has_onion or ((not other_has_onion) and (others_min_cost < own_min_cost)):
+                # Now find the *next* task that needs to be done, and do that
+    
```

Comment:
We shouldn't commit much deprecated & commented out code to overcooked_ai. This is another good reason to move most of your agents (and all this other code) to your project repo, so you can keep it a bit less organized till release.

Reply:
I've now moved my agents (GreedyHumanModel_pk and ToMModel) to my repo: human_ai_robustness.agents. This means that there isn't much depreciated/commented out code left in overcooked_ai_py/agents/agent.py, so it's much cleaner now :)